Set automatic update system for message function

root8000 opened the pull request:
# WHAT
Set automatic update system for message function
# WHY
必須機能のため

Commit:
Modify create.json.jbuilder(messages)

・見づらかったのでインデントを調整。
・投稿時間のフォーマットを統一。

Commit:
Create index.json.jbuilder(messages)

Commit:
Modify _message.html.haml

・チャット画面の自動更新のため、messageにidを付属

Commit:
Modify message.js

チャット画面自動更新のため編集
・各messageにidを付属
・今のグループチャット画面にいる間は５秒毎に自動更新するよう定義
・Ajaxの設定
・非同期で得たmessageのHTMLを作成し、チャット画面の最後尾に追加するよう定義

Commit:
Modify messages_controller.rb

最新messageのidよりも大きいidがあるかどうかで処理をわけるよう追記

Commit:
Modify message.js

不要な記述を削除

Commit:
modify message.js

・不要なコメントアウトを削除
・当初やろうとしていた最新messageだけを追加ではなく、全messageを描写し直すことで妥協したので、コメントを訂正。チャット数が多くなればなるほど負荷がかかると思うので、いつか妥協せずに取り組みたい。

taketnk commented on app/assets/javascripts/message.js:
```diff
@@ -53,4 +52,39 @@ $(function(){
+      insertHTML += buildHTML(message);
+    });
+    //作成したHTMLでメッセージ画面を再描写
+    $('.main-content__body__messages-list').html(insertHTML);
```
ここって.htmlで大丈夫でしたか？
ちょっと確認してみてください。

root8000 replied:
ご指摘ありがとうございます！再度確認したところ、やはり思っていた挙動と違うことをしていたので、諸々修正した後、この部分もhtmlではなくappendに書き換えました。

Commit:
Set pry-rails

必要になったので導入

Commit:
Modify create.json.jbuilder

createの際、idを取得しておらずindexで作成した各messageにidが付属出来ていなかったので修正

Commit:
Modify message controller

自動更新のアプローチ変更。コントローラでidの仕分けをしたかったが、ビューでやった方がうまく動いたので、コントローラでの記述を変更。役割的には本来はコントローラでやるべきだと思う。

Commit:
Modify message.js

自動更新のアプローチを変更。全messageのHTMLを作成し直すのではなく、新しいmessageだけHTMLを作成するよう記述。

Commit:
Modify message.js

resetContentだと不要な記述も多く意図しない挙動になってしまったため、スクロールに関してだけの記述に変更。

Commit:
Modify left_contents.scss

メッセージ送信アラートが表示されないことがある不具合を修正。

root8000 commented:
無事、意図する挙動になりました！！！！！！！
![default](https://user-images.githubusercontent.com/35862581/36627494-f22981c4-1986-11e8-9fdb-c532a60f5d09.gif)
